bundles/org.openhab.binding.mikrotik/src/main/java/org/openhab/binding/mikrotik/internal/handler/MikrotikBaseThingHandler.java

```diff
@@ -0,0 +1,239 @@
+                        executeCommand(channelUID, command);
+                    } catch (RuntimeException e) {
+                        logger.warn("Unexpected error handling command = {} for channel = {} : {}", command, channelUID,
+                                e.getMessage());
+                    }
+                }
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        cancelRefreshJob();
+        if (getVerifiedBridgeHandler() == null) {
+            updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "This thing requires a RouterOS bridge");
+            return;
+        }
+
+        Class<?> klass = (Class<?>) (((ParameterizedType) getClass().getGenericSuperclass())
+                .getActualTypeArguments()[0]);
+
+        C localConfig = (C) getConfigAs(klass);
+        if (localConfig == null) {
+            updateStatus(OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "Cannot obtain thing configuration");
+            return;
+        }
+        this.config = localConfig;
+
+        if (!config.isValid()) {
+            updateStatus(OFFLINE, CONFIGURATION_ERROR, String.format("%s is invalid", klass.getSimpleName()));
+            return;
+        }
+
+        updateStatus(ONLINE);
+    }
+
+    @Override
+    protected void updateStatus(ThingStatus status, ThingStatusDetail statusDetail, @Nullable String description) {
+        if (status == ONLINE || (status == OFFLINE && statusDetail == ThingStatusDetail.COMMUNICATION_ERROR)) {
+            scheduleRefreshJob();
+        } else if (status == OFFLINE
+                && (statusDetail == ThingStatusDetail.CONFIGURATION_ERROR || statusDetail == ThingStatusDetail.GONE)) {
+            cancelRefreshJob();
+        }
+
+        // update the status only if it's changed
+        ThingStatusInfo statusInfo = ThingStatusInfoBuilder.create(status, statusDetail).withDescription(description)
+                .build();
+        if (!statusInfo.equals(getThing().getStatusInfo())) {
+            super.updateStatus(status, statusDetail, description);
+        }
+    }
+
+    private void scheduleRefreshJob() {
+        synchronized (this) {
+            if (refreshJob == null) {
+                MikrotikRouterosBridgeHandler bridgeHandler = getVerifiedBridgeHandler();
+                if (bridgeHandler != null) {
+                    int refreshPeriod = bridgeHandler.getBridgeConfig().refresh;
+                    logger.debug("Scheduling refresh job every {}s", refreshPeriod);
+
+                    this.refreshCache = new ExpiringCache<>(Duration.ofSeconds(refreshPeriod),
+                            this::verifiedRefreshModels);
+                    refreshJob = scheduler.scheduleWithFixedDelay(this::scheduledRun, refreshPeriod, refreshPeriod,
+                            TimeUnit.SECONDS);
+                } else {
+                    logger.error("Cannot schedule refresh job since getVerifiedBridgeHandler() is null");
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. You could set the thing offline with an appropriate error message.

Reply:
Done